Library/Homebrew/formula_creator.rb

```diff
@@ -98,6 +98,7 @@ def write_formula!
           @desc = @github["description"]
           @homepage = @github["homepage"].presence || "https://github.com/#{@github["full_name"]}"
           @license = @github["license"]["spdx_id"] if @github["license"]
+          @mode = @github["language"].downcase.to_sym if @mode.nil?
```

Comment:
How confident are you that this is "correct" for the majority of repositories? It may be worth running this on e.g. all homebrew/core GitHub repositories to see how often it matches up

Reply:
It is better than default C++ template, which is guaranteed to miss on all non C++ projects.

I am pretty sure I don't have enough quota to check all homebrew formulas, but here is the prompt that can be modified to spill out the code to run.

--- 

Hey Qwen. Can you check language (mode) of all Homebrew formulas that are hosted on GitHub and say how many of them match `project["language"]` field that is returned form GitHub API?